packages/cozy-harvest-lib/src/components/hoc/withIntents.jsx

```diff
@@ -0,0 +1,38 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+
+import Intents from 'cozy-interapp'
+import { withClient } from 'cozy-client'
+
+/**
+ * @description HOC to provide intents singleton as prop
+ *
+ * @param  {Component} WrappedComponent
+ * @return {function} - Component that will receive intents as prop
+ */
+export const withIntents = WrappedComponent => {
+  class ComponentWithIntents extends Component {
+    static intents
+
+    constructor(props, context) {
+      super(props, context)
+      const { client } = props
+      this.intents = this.intents || new Intents({ client })
```

Comment:
Are you sure this works as intended? Since the class is redefined every time the HOC is used, I would think that `this.intents` is never truthy, but maybe I'm missing something?

Reply:
Well spotted, I missed the point that the class was redefined every time. This still works here because the only goal of the singleton is to avoid creating a new instance of Intents every time, but as you pointed out, it is eventually what it is done.

What solution do you prefer ?

1) Definining the `ComponentWithIntents` outside the HOC
2) Definining `intents` outside the HOC
3) Another solution I missed.

I prefer 1).

Reply:
1 would be cool but I don't think we can do it since we won't have access to WrappedComponent. But 2 is good as well 👍